client/app/cloud/project/compute/regions/regions.controller.js

```diff
@@ -0,0 +1,90 @@
+  }
+
+  $onInit() {
+    this.availableRegionToAdd = null;
+    this.initLoaders();
+  }
+
+  initLoaders() {
+    this.regions = this.ControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.OvhApiCloudProjectRegion.v6()
+        .query({ serviceName: this.serviceName })
+        .$promise
+        .then(regionIds => _.map(regionIds, region => this.RegionService.getRegion(region)))
+        .catch(error => this.ServiceHelper.errorHandler('cpci_add_regions_get_regions_error')(error)),
+    });
+    this.regions.load();
+    this.regionsByDatacenter = this.ControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.regions
+        .promise
+        .then(regions => this.VirtualMachineAddService.constructor.groupRegionsByDatacenter(
+          regions,
+        )),
+    });
+    this.regionsByDatacenter.load();
+    this.regionsByContinent = this.ControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.regionsByDatacenter
+        .promise
+        .then(regions => _.groupBy(regions, 'continent')),
+    });
+    this.regionsByContinent.load();
+
+    // load available regions
+    this.availableRegions = this.ControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.OvhApiCloudProjectRegion.AvailableRegions().v6()
+        .query({ serviceName: this.serviceName })
+        .$promise
+        .then(regionIds => _.map(regionIds, region => this.RegionService.getRegion(region.name)))
+        .catch(error => this.ServiceHelper.errorHandler('cpci_add_regions_get_available_regions_error')(error)),
+    });
+    this.availableRegions.load();
+    this.availableRegionsByDatacenter = this.ControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.availableRegions
+        .promise
+        .then(regions => this.VirtualMachineAddService.constructor.groupRegionsByDatacenter(
+          regions,
+        )),
+    });
+    this.availableRegionsByDatacenter.load();
+    this.availableRegionsByContinent = this.ControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.availableRegionsByDatacenter
+        .promise
+        .then(regions => _.groupBy(regions, 'continent')),
+    });
+    this.availableRegionsByContinent.load();
+  }
+
+  addRegions() {
+    this.CloudMessage.flushChildMessage();
+    this.addRegion = this.ControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.OvhApiCloudProjectRegion.v6()
+        .addRegion({ serviceName: this.serviceName },
+          { region: this.availableRegionToAdd.microRegion.code })
+        .$promise
+        .then(() => this.initLoaders())
+        .then(() => this.ServiceHelper.successHandler('cpci_add_regions_add_region_success')({ code: this.availableRegionToAdd.microRegion.code }))
+        .catch(error => this.ServiceHelper.errorHandler('cpci_add_regions_add_region_error')(error))
+        .finally(() => this.ControllerHelper.scrollPageToTop()),
+    });
+    this.addRegion.load();
```

Comment:
`return` ?

Reply:
not required

Reply:
Yes, this.addRegion.load() returns a promise but I don't use it. As you said it might be best practice to return it. We might get a situation to use it in the future. I will make this change.

Reply:
added return